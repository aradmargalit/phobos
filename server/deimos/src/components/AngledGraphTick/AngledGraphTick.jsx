import React from 'react';

export default function AngledGraphTick({ x, y, payload }) {
  return (
    <g transform={`translate(${x},${y})`}>
      <text
        x={0}
        y={0}
        dy={16}
        textAnchor="end"
        fill="#666"
        transform="rotate(-45)"
      >
        {payload.value}
      </text>
    </g>
  );
}
