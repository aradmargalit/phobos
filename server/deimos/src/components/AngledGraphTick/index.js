import AngledGraphTick from './AngledGraphTick';

export default AngledGraphTick;
