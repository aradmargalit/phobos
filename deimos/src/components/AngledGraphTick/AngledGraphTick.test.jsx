import { shallow } from 'enzyme';
import React from 'react';

import AngledGraphTick from './AngledGraphTick';

const generateComponent = (props) => {
  const defaultProps = {
    x: 1,
    y: 2,
    payload: { value: 'December 2018' },
  };
  const mergedProps = { ...defaultProps, ...props };

  return shallow(<AngledGraphTick {...mergedProps} />);
};

describe('<AngledGraphTick />', () => {
  it('renders correctly', () => {
    expect(generateComponent()).toMatchSnapshot();
  });
});
