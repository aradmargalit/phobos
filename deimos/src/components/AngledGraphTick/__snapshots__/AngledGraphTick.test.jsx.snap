// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AngledGraphTick /> renders correctly 1`] = `
<g
  transform="translate(1,2)"
>
  <text
    dy={16}
    fill="#666"
    textAnchor="end"
    transform="rotate(-45)"
    x={0}
    y={0}
  >
    December 2018
  </text>
</g>
`;
